sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/kinesis/enhancedfanout/EFOShardSubscribersPoolTest.java

```diff
@@ -153,27 +152,33 @@ public void poolReSubscribesWhenRecoverableErrorOccurs() throws Exception {
 
     pool = new EFOShardSubscribersPool(readSpec, kinesis, 1L);
     pool.start(initialCheckpoint);
-    List<KinesisRecord> actualRecords = waitForRecords(pool, 25);
-    assertEquals(18, actualRecords.size());
 
-    List<SubscribeToShardRequest> expectedSubscribeRequests =
-        ImmutableList.of(
+    assertThat(waitForRecords(pool, 18)).hasSize(18);
```

Comment:
I recommend using assertJ for tests. It makes it a lot easier to write meaningful assertions.